fondat/aws/cloudwatch.py

```diff
@@ -130,13 +130,46 @@ class CloudWatchMonitor:
     A monitor that stores all recorded measurements in CloudWatch.
     """
 
-    # now: just do one-to-one (immediately post the metric to AWS cloudwatch)
     # future: collect metrics, send in batches
 
     def __init__(self, client: Client, namespace: str):
         self.resource = cloudwatch_resource(client=client).namespace(namespace)
 
     async def record(self, measurement: Measurement):
         """Record a measurement."""
-        raise NotImplementedError
-        # resource.metrics.post(...)
+
+        if measurement.type == "counter":
+            m = Counter(timestamp=measurement.timestamp)
+            m.record(measurement.value)
+            metric = Metric(
+                name=measurement.tags["name"],
+                dimensions={"Name": measurement.type, "Value": str(m.value)},
+                timestamp=measurement.timestamp,
+                value=float(m.value),
+                unit="Count",
+            )
+        elif measurement.type == "gauge":
+            m = Gauge(timestamp=measurement.timestamp)
```

Comment:
I think this is fine for now. Count will always be 1 (and min/max/sum will all be the same), but once we adapt this to collect multiple metrics and publish to CloudWatch, will become more useful. I'll add helpers in `fondat.monitoring` to support that.